src/expire.c

```diff
@@ -610,9 +613,10 @@ void expireGenericCommand(client *c, long long basetime, int unit) {
     }
 
     /* LT option is set, check current expiry */
-    if (flag & FLAG_LT) {
+    if (flag & EXPIRE_LT) {
         current_expire = getExpire(c->db, key);
-        if (when >= current_expire) {
+        /* there is an expiry on the key and it's not less than current */
+        if (when >= current_expire && current_expire > -1) {
```

Comment:
```suggestion
        /* When current_expire -1, we consider it as infinite TTL,
         * But 'when' can still be negative at this point, so if there is
         * an expiry on the key and it's not less than current, we fail the GT. */
        if (current_expire != -1 && when >= current_expire) {
```

Reply:
the check here is for `LT`. Perhaps it's a typo with `we fail the GT`?  So does the `LT` below.

Reply:
yes. typo.